airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/job.py

```diff
@@ -306,6 +310,15 @@ def job_check_for_errors(self, response: requests.Response) -> Union[AirbyteTrac
                 f"Couldn't check the `response` for `errors`, status: {response.status_code}, response: `{response.text}`. Trace: {repr(e)}."
             )
 
+    def job_one_time_retry_error(self, response: requests.Response, exception: Exception) -> Optional[requests.Response]:
+        if not self.one_time_error_retried:
+            request = response.request
+            self.logger.info(f"Stream: `{self.stream_name}`, retrying `Bad Request`: {request.body}. Error: {repr(exception)}.")
```

Comment:
Based on [this line](https://github.com/airbytehq/airbyte/pull/37468/files#diff-69c119a9dc0a2f8a93bc200b41dfe1343d3ac94f117698a9442f30c5e2cadd60R340), it could also be an unknown error, right?

Reply:
Yes, we will try to retry Unknown error, at least once, as well.